CMakeLists.txt

```diff
@@ -368,7 +368,9 @@ if(WITH_CORE)
     message (SEND_ERROR "sqlite3 dependency was not found!")
   endif()
 
-  find_package(Protobuf REQUIRED)   # for decoding of vector tiles in MVT format
+  find_package(Protobuf CONFIG REQUIRED)  # for decoding of vector tiles in MVT format
+  find_package(Protobuf) # for protobuf_generate_cpp
```

Comment:
it used two way to find Protobuf package, Config Mode with protobuf's files and Module Mode with cmake's file.

[Protobuf examples](https://github.com/protocolbuffers/protobuf/blob/main/examples/CMakeLists.txt) show how to find_package Config Mode with legacy support for "protobuf_generate_cpp", or the another way without legacy support, use "protobuf_generate", I have a [commit](https://github.com/wulu90/QGIS/commit/05989e197e7bbb142c893929f969f6c91529d5d1) in my forked repo .

The CI use ubuntu-latest(22.04) , the version of libprotobuf-dev in ubuntu 22.04 is 3.12.4, this package didn't have cmake-config.cmake files. [package files list is here](https://packages.ubuntu.com/jammy/amd64/libprotobuf-dev/filelist) . I think this is the same reason why MacOS CI was break.

Reply:
thanks for your input @wulu90 :+1: